Automattic/jetpack
# JS Components: Convert ProductPrice component to TypeScript

#### Changes proposed in this Pull Request:
* This PR converts `ProductPrice` component to TypeScript and cleans up the propTypes.
* It also converts the `ProductPrice` stories to TS

#### Other information:


#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:
* Checkout the PR
* Run `cd projects/js-packages/storybook && pnpm run storybook:dev`
* Confirm that all the stories of `ProductPrice` component work as expected
* Check the consumers of `ProductPrice` and related components and confirm that they work as expected
* Check for any regression


<img width="611" alt="Screenshot 2022-07-01 at 5 24 55 PM" src="https://user-images.githubusercontent.com/18226415/176889721-be747663-71d4-4734-a106-89f9e5fd5fe8.png">


---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1202276314263991
